pkg/volume/cinder/attacher.go

```diff
@@ -234,33 +271,8 @@ func (attacher *cinderDiskAttacher) WaitForAttach(spec *volume.Spec, devicePath
 		return "", fmt.Errorf("WaitForAttach failed for Cinder disk %q: devicePath is empty.", volumeID)
 	}
 
-	ticker := time.NewTicker(checkSleepDuration)
-	defer ticker.Stop()
-	timer := time.NewTimer(timeout)
-	defer timer.Stop()
-
-	for {
-		probeAttachedVolume()
-		select {
-		case <-ticker.C:
-			glog.V(5).Infof("Checking Cinder disk %q is attached.", volumeID)
-			probeAttachedVolume()
-			if !attacher.cinderProvider.ShouldTrustDevicePath() {
-				// Using the Cinder volume ID, find the real device path (See Issue #33128)
-				devicePath = attacher.cinderProvider.GetDevicePath(volumeID)
-			}
-			exists, err := volumeutil.PathExists(devicePath)
-			if exists && err == nil {
-				glog.Infof("Successfully found attached Cinder disk %q at %v.", volumeID, devicePath)
-				return devicePath, nil
-			} else {
-				// Log an error, and continue checking periodically
-				glog.Errorf("Error: could not find attached Cinder disk %q (path: %q): %v", volumeID, devicePath, err)
-			}
-		case <-timer.C:
-			return "", fmt.Errorf("Could not find attached Cinder disk %q. Timeout waiting for mount paths to be created.", volumeID)
-		}
-	}
+	// Using exponential backoff instead of linear
+	return attacher.waitProbeVolume(devicePath, volumeID)
```

Comment:
previously, the `timeout` parameter was taken into account and now it is not. The max. wait time is 1023 seconds now.